src/File.php

```diff
@@ -485,6 +484,49 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+
+    /**
+     * Check if the File has a CanViewType set or inherited at "LoggedInUsers or above"
+     *
+     * @return bool
+     */
+    public function hasRestrictedAccess(): bool
+    {
+        return $this->hasRestrictedPermissions($this);
+    }
+
+    /**
+     * @param File $file
+     * @return bool
+     */
+    private function hasRestrictedPermissions(File $file): bool
+    {
+        $canViewType = $file->CanViewType;
+        if (in_array($canViewType, [InheritedPermissions::LOGGED_IN_USERS, InheritedPermissions::ONLY_THESE_USERS])) {
+            return true;
+        } elseif ($canViewType == InheritedPermissions::INHERIT) {
+            if ($file->ParentID != 0) {
+                $parent = $file->Parent();
+                if ($parent->exists()) {
+                    return $this->hasRestrictedPermissions($parent);
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Whether the file was originally uploaded in a userform submission by a FileField
+     *
+     * @see UserFormFileExtension::updateIsUserFormUpload
+     */
+    public function isUserFormUpload(): bool
+    {
+        $value = false;
+        $this->extend('updateIsUserFormUpload', $value);
+        return $value;
+    }
```

Comment:
This should be more generic - something like `isTrackedFormUpload()` - so that we aren't referencing a completely optional module in a core one, and so that it's more sensible for other modules / developers to make use of in the future.

Reply:
I sort of agree, though at the same time it's pretty baked into the admin/asset-admin JS that we're looking a userform upload specifically

In theory this could be extended later to also cover 'regular form' uploads, however we've already decided early in this piece that's out of scope as you `[cannot retroactively identify uploads](https://github.com/silverstripe/silverstripe-userforms/issues/950#issuecomment-616320644)`, so it's pretty unlikely this will ever get extended

Reply:
I still think it's worth naming generically - our tooltips for this feature in the CMS make no reference to userforms, and I could easily see developers adopting this API in custom forms on their own projects. We should try not to reference downstream modules in core to avoid confusion and implied coupling.

Reply:
There is a scenario where:
- a brand new website is built with no existing data
- therefore don't need to worry about half the form uploads having missing data - which causes the icon to be displayed inconsistently (this is much worse than no icon at all as your CMS if full of false negatives.)
- developers add their custom logic to MyApp\FileExtensions::updateIsTrackedFormUpload() that somehow registers custom FormField uploads

However, there's still some issues with this:
- We're providing a method to developers to introduce missing data, basically any future dev work (post initial build, think years later) involving a form with uploads MUST do something with this extension method, and when someone forgets they'll end up with same inconsistent icon experience.
- We're essentially highjacking the userform submission icon to do something that it wasn't designed to do.  The reason the tooltips say 'Form submission' instead of 'UserForm submissions' is PUX's view that most cms-users won't know what a 'UserForm' is.  We had a long discussion of the tooltip saying 'Form' vs 'UserForm'.
- This is scope creep.  We specifically excluded catering for custom form uploads and we made no mention of extensibility when doing the specs for this work.

Therefore, my view is still that we should keep this as is.

Reply:
```
# This doesn't work, doesn't read draft site:
return Versioned::withVersionedMode(function() {
    Versioned::set_reading_mode('Stage.' . Versioned::DRAFT);
    return Member::actAs(null, function() {
        return !$this->canView();
    });
});

# This works, fast performance, but is it risky?
$default = Versioned::get_draft_site_secured();
Versioned::set_draft_site_secured(false);
$value = Member::actAs(null, function() {
    return !$this->canView();
});
Versioned::set_draft_site_secured($default);
return $value;

2 runs:
11,232
35,774
```

Reply:
- Arguably this will be 'inconsistent' whether we explicitly make this API available to developers or not, thanks to the broad description used in the tooltips. By naming this as a pattern suitable for broader use, we at least give them the option to make it more consistent if they have the scope / ability.
- @sachajudd / @clarkepaul - Do you see the form submission indicator as being exclusive to UserForms, or would you be comfortable with it being used more broadly in projects where developers opted to apply it to their own Forms?
- I don't think this change fits the term 'scope creep'; if we went through with it, it'd require a batch of renaming and nothing more. We're not catering for it in terms of actually supporting it - we're catering for it in terms of allowing the possibility of it in the future.

Once this ships, it's locked in - even if we were to leave it undocumented for now and then seek to expose it more widely later, we wouldn't be able to adjust the naming until CMS 5 because it introduces public API surface.

Sensible extensibility is an NFR for features in the Open Source project.

Reply:
> we wouldn't be able to adjust the naming until CMS 5 because it introduces public API surface.
Agree that this isn't great. Why don't we just batch rename it to 'isTrackedFormUpload' and then we actually have the option of allowing this possibility in the future? I don't see why there is such a big push back on this.

Reply:
I agree with Garion - please don't couple core modules to optional ones

Reply:
My suggestion would be to find out the reason why this method has to exist, e.g. because we want to distinguish userform uploads differently from regular assets in the assets area (for example, for some reason). If that's the example reason here then design your API around providing extensibility for that - e.g. `getIcon(): string` and have an extension point inside it.

Reply:
Have updated naming on all the PR's to `isTrackedFormUpload`